pkg/db/client.go

```diff
@@ -138,40 +139,76 @@ func (c *ClientSqlx) GetResult(query string, args ...interface{}) (*Result, erro
 	return &out, err
 }
 
-func (c *ClientSqlx) Exec(query string, args ...interface{}) (sql.Result, error) {
+func (c *ClientSqlx) Exec(ctx context.Context, query string, args ...interface{}) (sql.Result, error) {
 	c.logger.Debug("> %s %q", query, args)
 
-	return c.db.Exec(query, args...)
+	return c.db.ExecContext(ctx, query, args...)
 }
 
-func (c *ClientSqlx) Prepare(query string) (*sql.Stmt, error) {
-	return c.db.Prepare(query)
+func (c *ClientSqlx) Prepare(ctx context.Context, query string) (*sql.Stmt, error) {
+	return c.db.PrepareContext(ctx, query)
 }
 
-func (c *ClientSqlx) Query(query string, args ...interface{}) (*sql.Rows, error) {
+func (c *ClientSqlx) Query(ctx context.Context, query string, args ...interface{}) (*sql.Rows, error) {
 	c.logger.Debug("> %s %q", query, args)
 
-	return c.db.Query(query, args...)
+	return c.db.QueryContext(ctx, query, args...)
 }
 
-func (c *ClientSqlx) QueryRow(query string, args ...interface{}) *sql.Row {
-	return c.db.QueryRow(query, args...)
+func (c *ClientSqlx) QueryRow(ctx context.Context, query string, args ...interface{}) *sql.Row {
+	return c.db.QueryRowContext(ctx, query, args...)
 }
 
-func (c *ClientSqlx) Queryx(query string, args ...interface{}) (*sqlx.Rows, error) {
+func (c *ClientSqlx) Queryx(ctx context.Context, query string, args ...interface{}) (*sqlx.Rows, error) {
 	c.logger.Debug("> %s %q", query, args)
 
-	return c.db.Queryx(query, args...)
+	return c.db.QueryxContext(ctx, query, args...)
 }
 
-func (c *ClientSqlx) Select(dest interface{}, query string, args ...interface{}) error {
+func (c *ClientSqlx) Select(ctx context.Context, dest interface{}, query string, args ...interface{}) error {
 	c.logger.Debug("> %s %q", query, args)
 
-	return c.db.Select(dest, query, args...)
+	return c.db.SelectContext(ctx, dest, query, args...)
 }
 
-func (c *ClientSqlx) Get(dest interface{}, query string, args ...interface{}) error {
+func (c *ClientSqlx) Get(ctx context.Context, dest interface{}, query string, args ...interface{}) error {
 	c.logger.Debug("> %s %q", query, args)
 
-	return c.db.Get(dest, query, args...)
+	return c.db.GetContext(ctx, dest, query, args...)
+}
+
+func (c *ClientSqlx) BeginTx(ctx context.Context, ops *sql.TxOptions) (*sql.Tx, error) {
+	c.logger.Debug("start tx")
+
+	return c.db.BeginTx(ctx, ops)
+}
+
+func (c *ClientSqlx) OnTx(ctx context.Context, ops *sql.TxOptions, do func(ctx context.Context, tx *sql.Tx) error) error {
```

Comment:
Please revisit the error handling in this function:
- name the return value and use multierror.Append to return both errors should there be an error during rollback
- check your error handling in this function - returning no error in case of `do` returning an error looks extremely fishy
- the last condition looks quite redundant - both branches return `err`